app/scripts/metamask-controller.js

```diff
@@ -5977,6 +5984,21 @@ export default class MetamaskController extends EventEmitter {
               this.networkController.removeNetworkConfiguration.bind(
                 this.networkController,
               ),
+            requestPermissionApprovalForOrigin: async (permissions) => {
+              const id = nanoid();
+              return this.approvalController.addAndShowApprovalRequest({
+                id,
+                origin,
+                requestData: {
+                  metadata: {
+                    id,
+                    origin,
+                  },
+                  permissions,
+                },
+                type: MethodNames.requestPermissions,
+              });
+            },
```

Comment:
Can this be DRYed? https://github.com/MetaMask/metamask-extension/pull/26483/files#diff-6fbff2cfe97ac01b77296ef2122c7e0a5b3ff6a84b584b4d1a87482f35eea3d6R5708

Reply:
and here: https://github.com/MetaMask/metamask-extension/pull/26483/files#diff-6fbff2cfe97ac01b77296ef2122c7e0a5b3ff6a84b584b4d1a87482f35eea3d6R6145

Reply:
done https://github.com/MetaMask/metamask-extension/pull/26483/commits/b49f97c8b90d6e3802b21ca2caa98cbcff1365d9